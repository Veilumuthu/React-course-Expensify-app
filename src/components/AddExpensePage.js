import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses'

const AddExpensePage = ({startAddExpense, history}) => {
    const onSubmit = (expense) => {
        startAddExpense(expense)
        history.push('/');
    }
    return (
        <div>
            <div className = "page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Add Expense</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm onSubmit={onSubmit} />
            </div>
        </div>
    )
}
// export class AddExpensePage extends React.Component {

//     render() {
//         return (

//         )
//     }
// }

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})
export default connect(undefined, mapDispatchToProps)(AddExpensePage);