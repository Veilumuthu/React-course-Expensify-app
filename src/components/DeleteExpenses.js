import React from 'react';
import {startDeleteExpense} from '../actions/expenses';
import {connect} from 'react-redux';

class DeleteExpenses extends React.Component {
    onDelete = (expense) => {
        this.props.startDeleteExpense(expense)
    }
    render () {
        return (
            <div className="content-container">
                <button className="button--delete" onClick= {this.onDelete}>Delete Expenses</button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startDeleteExpense: (expense) => dispatch(startDeleteExpense(expense))
})
export default connect(undefined, mapDispatchToProps)(DeleteExpenses);