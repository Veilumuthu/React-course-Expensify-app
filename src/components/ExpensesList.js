import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectorExpense from '../selectors/expenses'
// for tesing
export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map((expense) => {
            return <ExpenseListItem  key = {expense.id} {...expense}/>
        })}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectorExpense(state.expenses, state.filters)
    };
}
export default connect(mapStateToProps)(ExpenseList);
