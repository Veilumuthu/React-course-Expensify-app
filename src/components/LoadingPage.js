import React from 'react';

const Loadingpage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/771.gif" />
    </div>
);

export default Loadingpage;