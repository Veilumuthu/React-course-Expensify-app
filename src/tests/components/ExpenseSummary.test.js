import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render expenses summary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235123457} />)
    expect(wrapper).toMatchSnapshot();
})