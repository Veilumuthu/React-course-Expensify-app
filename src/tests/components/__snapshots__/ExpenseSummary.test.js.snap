// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     
    $2,351,234.57
  </h1>
</div>
`;

exports[`should correctly render expenses summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     
    $2.35
  </h1>
</div>
`;
